spring/src/main/java/org/axonframework/spring/config/SpringComponentRegistry.java

```diff
@@ -157,23 +158,28 @@ public <C> ComponentRegistry registerDecorator(@Nonnull DecoratorDefinition<C, ?
     }
 
     @Override
-    public boolean hasComponent(@Nonnull Class<?> type) {
+    public boolean hasComponent(@Nonnull Class<?> type,
+                                @Nullable String name,
+                                @Nonnull SearchScope searchScope) {
         // Checks both the local Components as the BeanFactory,
         //  since the ConfigurationEnhancers act before component registration with the Application Context.
-        return components.contains(new Component.Identifier<>(type, null))
-                || beanFactory.getBeanNamesForType(type).length > 0;
+        return switch (searchScope) {
+            case ALL -> components.contains(new Component.Identifier<>(type, name)) || contextHasComponent(type, name);
+            case CURRENT -> components.contains(new Component.Identifier<>(type, name));
+            case ANCESTORS -> contextHasComponent(type, name);
```

Comment:
If we execute `hasComponent(something, something, SearchScope.ANCESTORS` while on the main configurer, does this mean that this is also the current?

Reply:
My concern is that for Spring `ANCESTORS` acts like `ALL`, right?

Reply:
This depends a little on the life cycle of the `ComponentRegistry`. If the `Components` collection from **this** registry have been added to the `ApplicationContext`, then yes.
Because `ANCESTORS` moves to the `BeanFactory`, and thus retrieves any registered beans.

Concerning `ConfigurationEnhancers`, which will typically invoke these has-components methods, then we are certain the current `Components` have not been registered yet.
This has to do with the ordering of the `SpringComponentRegistry#initialize`; enhancers are invoked first, then decorators, and only **then** do we registers the `Components` entries with the Application Context.